### PR TITLE
Add Node::hasChildren

### PR DESCRIPTION
**Pull request by szepeviktor**
When PHPStan says _Construct empty() is not allowed. Use more strict comparison._

### CODE REVIEW

File: src/Mapper/Printer/HtmlTreePrinter.php [hunk 1]
```diff
@@ -99,7 +99,7 @@ public function print(Node $node): string
     {
         $value = $node->isValid() ? $node->value() : null;
 
-        if (! empty($node->children())) {
+        if ($node->hasChildren()) {
```
Comment (Ocramius) [inline src/Mapper/Printer/HtmlTreePrinter.php:102]: Tests are really really really needed for this sort of change :D

File: src/Mapper/Tree/Node.php [hunk 2]
```diff
@@ -135,6 +135,11 @@ public function children(): array
         return $this->children;
     }
 
+    public function hasChildren(): bool
```
Comment (Ocramius) [inline src/Mapper/Tree/Node.php:138]: Tests are really really really needed for this sort of change :pray:

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/Mapper/Printer/HtmlTreePrinter.php (hunk 1 above)
Comment (romm) [reply src/Mapper/Printer/HtmlTreePrinter.php:102]: Damn, this code should never have landed in the public repo :D

I'll probably delete this class anyway, and re-work on it later when other major issues are resolved.

Comment (szepeviktor) [reply src/Mapper/Printer/HtmlTreePrinter.php:102]: All right I close this repo.
Now I don't need to learn how to write tests 🙃

**Pull request closed without merging**